src/dbos-runtime/config.ts

```diff
@@ -142,52 +139,191 @@ function retrieveApplicationName(configFile: ConfigFile): string {
     }
   }
-  configFile.database.hostname = process.env.DBOS_DBHOST || configFile.database.hostname || 'localhost';
-  const dbos_dbport = process.env.DBOS_DBPORT ? parseInt(process.env.DBOS_DBPORT) : undefined;
-  configFile.database.port = dbos_dbport || configFile.database.port || 5432;
-  configFile.database.username = process.env.DBOS_DBUSER || configFile.database.username || 'postgres';
-  configFile.database.password =
-    process.env.DBOS_DBPASSWORD || configFile.database.password || process.env.PGPASSWORD || 'dbos';
-
-  let databaseName: string | undefined = configFile.database.app_db_name;
-  const appName = retrieveApplicationName(configFile);
-  if (globalParams.appName === '') {
-    globalParams.appName = appName;
-  }
-  if (databaseName === undefined) {
-    databaseName = appName.toLowerCase().replaceAll('-', '_').replaceAll(' ', '_');
-    if (databaseName.match(/^\d/)) {
-      databaseName = '_' + databaseName; // Append an underscore if the name starts with a digit
+
+  let connectionString = undefined;
+  let databaseName = undefined;
+
+  // If a database_url is found, parse it to backfill the poolConfig
+  if (configFile.database_url) {
+    const url = new URL(configFile.database_url);
+    // If in debug mode, apply the debug overrides
+    if (isDebugMode) {
+      if (process.env.DBOS_DBHOST) {
+        url.hostname = process.env.DBOS_DBHOST;
+      }
+      if (process.env.DBOS_DBPORT) {
+        url.port = process.env.DBOS_DBPORT;
+      }
+      if (process.env.DBOS_DBUSER) {
+        url.username = process.env.DBOS_DBUSER;
+      }
+      if (process.env.DBOS_DBPASSWORD) {
+        url.password = process.env.DBOS_DBPASSWORD;
+      }
+      configFile.database_url = url.toString();
+    }
+    connectionString = configFile.database_url;
+
+    // From here backfill the poolConfig
+
+    const queryParams = Object.fromEntries(url.searchParams.entries());
+
+    // Validate required fields
+    const missingFields: string[] = [];
+    if (!url.username) missingFields.push('username');
+    if (!url.password) missingFields.push('password');
+    if (!url.hostname) missingFields.push('hostname');
+
+    if (missingFields.length > 0) {
+      throw new Error(`Invalid database URL: missing required field(s): ${missingFields.join(', ')}`);
+    }
+
+    // Create database config object
+    configFile.database = {
+      hostname: url.hostname,
+      port: url.port ? parseInt(url.port, 10) : 5432,
+      username: url.username,
+      password: url.password,
+      app_db_name: url.pathname.substring(1) || undefined,
+      ssl: queryParams['sslmode'] === 'require' || queryParams['sslmode'] === 'verify-full',
+      ssl_ca: queryParams['sslrootcert'] || undefined,
+      connectionTimeoutMillis: queryParams['connect_timeout']
+        ? parseInt(queryParams['connect_timeout'], 10) * 1000
+        : 3000,
+    };
+
+    databaseName = configFile.database.app_db_name;
```

Comment:
What is this? Why do we need it if we're getting a connection string passed in?

Reply:
Oh, I see, you want the database to be optional in the connection string.

Reply:
Recommendation is to delete this code and make the database required. Will make things much easier in the future.

Reply:
I had this train of thoughts earlier. I was worried about 1-week backward compatibility with people not providing a dbname.